src/Microsoft.Management.Configuration/ConfigurationProcessor.cpp

```diff
@@ -626,6 +627,59 @@ namespace winrt::Microsoft::Management::Configuration::implementation
         return *result;
     }
 
+    Configuration::GetAllConfigurationUnitSettingsResult ConfigurationProcessor::GetAllUnitSettings(const ConfigurationUnit& unit)
+    {
+        THROW_HR_IF(E_NOT_VALID_STATE, !m_factory);
+        return GetAllUnitSettingsImpl(unit);
+    }
+
+    Windows::Foundation::IAsyncOperation<Configuration::GetAllConfigurationUnitSettingsResult> ConfigurationProcessor::GetAllUnitSettingsAsync(const ConfigurationUnit& unit)
+    {
+        THROW_HR_IF(E_NOT_VALID_STATE, !m_factory);
+
+        auto strong_this{ get_strong() };
+        ConfigurationUnit localUnit = unit;
+
+        co_await winrt::resume_background();
+
+        co_return GetAllUnitSettingsImpl(localUnit, { co_await winrt::get_cancellation_token() });
+    }
+
+    Configuration::GetAllConfigurationUnitSettingsResult ConfigurationProcessor::GetAllUnitSettingsImpl(
+        const ConfigurationUnit& unit,
+        AppInstaller::WinRT::AsyncCancellation cancellation)
+    {
+        auto threadGlobals = m_threadGlobals.SetForCurrentThread();
+
+        IConfigurationSetProcessor setProcessor = m_factory.CreateSetProcessor(nullptr);
+        auto result = make_self<wil::details::module_count_wrapper<implementation::GetAllConfigurationUnitSettingsResult>>();
+        auto unitResult = make_self<wil::details::module_count_wrapper<implementation::ConfigurationUnitResultInformation>>();
+        result->ResultInformation(*unitResult);
+
+        cancellation.ThrowIfCancelled();
+
+        IGetAllSettingsConfigurationUnitProcessor unitProcessor;
+        if (setProcessor.CreateUnitProcessor(unit).try_as<IGetAllSettingsConfigurationUnitProcessor>(unitProcessor))

```

Comment:
I don't think that we should return a default constructed result if the processor doesn't support export; it should be an error and a specific `HRESULT`.  Probably a new one like ...`_NOT_SUPPORTED_BY_PROCESSOR` or some variant.

Alternately, we could add another field to the result object, but I think the error is cleaner.

This currently makes it seem like there were no results instead of a clear indication that you simply cannot do that.